libbeat/processors/registry.go

```diff
@@ -0,0 +1,28 @@
+package processors
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type ProcessorInterface interface {
+	Process(event common.MapStr) (common.MapStr, error)
+	String() string
+}
+
+type Constructor func(config common.Config) (ProcessorInterface, error)
```

Comment:
Normally we pass config by pointer.

Today I learned passing config by pointer and by value is exactly equivalent. All modifications to config will still be visible, due to state is held by references to other objects. In this regard common.Config behaves like map[string]interface{}. It's funny how I wasn't aware of it :)